src/puppetlabs/puppetdb/middleware.clj

```diff
@@ -286,10 +286,30 @@
           (app (assoc req :body-string payload))
           (http/error-response (str "Missing required parameter 'payload'")))
         "application/json"
```

Comment:
It seems like this function and https://github.com/puppetlabs/puppetdb/pull/1807/files#diff-d82b3486b4cd22472cd542b7773086e9L24 could/should probably be combined into 1 function